web/modules/custom/dpl_event/src/PriceFormatter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\dpl_event;
+
+use Brick\Math\BigDecimal;
+use Drupal\Core\StringTranslation\TranslationInterface;
+
+/**
+ * Formats prices according to local rules.
+ *
+ * This is important as we do not rely on fields built for prices. Instead we
+ * have double fields with values as strings.
+ */
+class PriceFormatter {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    protected TranslationInterface $translation,
+  ) {}
```

Comment:
isnt it a mistake that this is empty?
shouldnt it contain something like

```php
$this->translation = $translation;
```
?

Reply:
This is intentional [constructor property promotion](https://php.watch/versions/8.0/constructor-property-promotion).